src/common/history.c

```diff
@@ -1206,6 +1206,58 @@ GList *dt_history_duplicate(GList *hist)
   return result;
 }
 
+const char *dt_hash_history_compute_from_db(int32_t imgid)
+{
+  if(imgid == -1) return NULL;
+
+  GChecksum *checksum = g_checksum_new(G_CHECKSUM_MD5);
+
+  // get history
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
+                              "SELECT operation, op_params, blendop_params"
+                              " FROM main.history"
+                              " WHERE imgid = ?1 AND enabled = 1"
+                              " ORDER BY num",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, imgid);
+
+  while(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    char *buf = (char *)sqlite3_column_text(stmt, 0);
+    int32_t params_len = sqlite3_column_bytes(stmt, 0);
+    if(buf) g_checksum_update(checksum, (const guchar *)buf, params_len);
+    buf = (char *)sqlite3_column_text(stmt, 1);
```

Comment:
This one should be sqlite3_column_blob() and not sqlite3_column_text()

Reply:
all wrong ...